src/Macro.php

```diff
@@ -40,8 +40,16 @@ protected function initPhpDoc($method)
         // Add macro parameters if they are missed in original docblock
         if (!$this->phpdoc->hasTag('param')) {
             foreach ($method->getParameters() as $parameter) {
-                $type = $parameter->hasType() ? $parameter->getType()->getName() : 'mixed';
-                $type .= $parameter->hasType() && $parameter->getType()->allowsNull() ? '|null' : '';
+                $reflectionType = $parameter->getType();
+
+                $type = $this->concatReflectionTypes($reflectionType);
+
+                /** @psalm-suppress UndefinedClass */
+                if ($reflectionType && !$reflectionType instanceof \ReflectionUnionType && $reflectionType->allowsNull()) {
```

Comment:
> \ReflectionUnionType

I assume in PHP < 8 this will "just work", because a non-existent class check in `instanceof` will not trigger any problems?

Reply:
Correct!